mp/src/game/shared/hl2mp/hl2mp_weapon_parse.cpp

```diff
@@ -73,6 +78,10 @@ void CHL2MPSWeaponInfo::Parse( KeyValues *pKeyValuesData, const char *szWeaponNa
 		m_angVMAimAngOffset[YAW] = pAimOffset->GetFloat("yaw", 0);
 		m_angVMAimAngOffset[ROLL] = pAimOffset->GetFloat("roll", 0);
 	}
+
+	// Get CycleTime AKA Fire-rate
+	m_flCycleTime = pKeyValuesData->GetFloat("CycleTime", 0.0f);
+	Assert(m_flCycleTime != 0.0f);
```

Comment:
We're hitting this assertion for HL2MP weapons. One solution would be removing the HL guns entirely, but maybe we wanna keep them for future modding capability by users?

Perhaps something like:
```cpp
#if _DEBUG
	if (V_strcmp(pKeyValuesData->GetString("printname"), "#HL2") != 0)
	{
		Assert(m_flCycleTime != 0.0f);
	}
#endif
```
to skip asserting the HL guns.

Reply:
This PR: https://github.com/NeotokyoRebuild/neo/pull/192 already removed a lot of HL guns, although a few remains in code but just prevented from spawning since they're rather tied to the codebase verses the others. Nevertheless will add the if statement anyway.